### ISSUE TITLE
[bug] Images with spaces fail to load

### ISSUE THREAD

**Issue by minpeter** [remcohaszing/remark-mdx-images#11]
hello

This plugin is very useful for saving images by folder.
However, if there is a space in the image file name, the following error occurs.
![image](https://github.com/remcohaszing/remark-mdx-images/assets/62207008/1582e8a9-d013-4d68-ba3b-ac3503192ff8)

The syntax I used to load the image is as follows:
```mdx
![](./images/mdx-logo%20copy.png)
```

After passing through the plugin, it appears to change as shown below.

```jsx
import mdxLogoCopy from "./images/mdx-logo%20copy.png";

<img src={mdxLogoCopy} />
```

If the plugin result is as follows, it is expected to operate normally. (It works if you enter it directly.)

```jsx
import mdxLogoCopy from "./images/mdx-logo copy.png";

<img src={mdxLogoCopy} />
```

Minor change to change the space from %20 to space ( )

I am currently building a blog using next.js and mdx-bunlder.

This can be easily solved by concatenating the image name with -. However, if this seems like a bug, please fix it.

thank :)

**Comment by remcohaszing** [comment]
I read this issue and at first thought it made perfect sense, so I started working on it.

The solution seems simple. Just decode it. But then I started thinking further. Imports are URLs, not paths. They accept query parameters for example. Those shouldn’t be encoded.

Also Node.js supports importing using the encoded URL.

```
$ cat 'a b.js'
console.log('hi')
$ node
Welcome to Node.js v18.18.2.
Type ".help" for more information.
> await import('./a%20b.js')
hi
[Module: null prototype] {  }
>
```

Apparently there is no need to decode. This isn’t a bug in `remark-mdx-images`, but somewhere else. Looking at your stack trace, this looks like a bug in esbuild.
